core/src/main/java/org/fao/geonet/kernel/mef/Importer.java

```diff
@@ -598,6 +601,52 @@ public static void importRecord(String uuid,
 
     // --------------------------------------------------------------------------
 
+    /**
+     * Applies a xslt process to update thumbnail links if apply.
+     *
+     * GeoNetwork versions prior to 3.2.0 stored in the thumbnail the name of the file,
+     * but since 3.2 is required a full url to the service returning the thumbnail.
+     *
+     * This process "fixes" the thumbnail links in metadata files created
+     * with versions previous to 3.2.0.
+     *
+     * @param context   ServiceContext.
+     * @param schema    Metadata schema.
+     * @param md        Metadata to process.
+     *
+     * @return Updated metadata.
+     *
+     * @throws Exception    Xslt process exception.
+     */
+    public static Element updateThumbnailLinks(final ServiceContext context,
+                                               final String schema,
+                                               final Element md)  throws Exception {
+
+        GeonetContext gc = (GeonetContext) context.getHandlerContext(Geonet.CONTEXT_NAME);
+        DataManager dm = gc.getBean(DataManager.class);
+
+        //--- do an XSL  transformation
+        Path styleSheetPath = dm.getSchemaDir(schema).resolve(Geonet.File.UPDATE_THUMBNAIL_LINKS);
```

Comment:
The issue is that can't use directly ```MetadataResourceDatabaseMigration.updateMetadataResourcesLink``` as will create a circular reference between ```core``` and ```services``` modules. 

Will check to refactor the code to manage this.